### PR TITLE
Refactor prop-types

### PR DESCRIPTION
**Pull request by tacigar**
* Collect the definitions of proptypes in various files into one file and use them.
* Fix type bug of limit condition component.

### CODE REVIEW

File: zipkin-lens/src/prop-types/index.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import PropTypes from 'prop-types';
```
Comment (tacigar) [inline zipkin-lens/src/prop-types/index.js:1]: This file has all prop-type definitions of spans and traces.

File: zipkin-lens/src/prop-types/index.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  timestamp: PropTypes.number.isRequired,
+  endpoint: PropTypes.string.isRequired,
+  relativeTime: PropTypes.string.isRequired,
+});
+
+export const spanAnnotationsPropTypes = PropTypes.arrayOf(
+  spanAnnotationPropTypes,
+);
+
+export const detailedSpanPropTypes = PropTypes.shape({
```
Comment (codefromthecrypt) [inline zipkin-lens/src/prop-types/index.js:25]: maybe add a TODO as while I think your assessment is correct based on span row conversion, the underlying data only requires trace and span ID. We need to revisit things like this once UX is working as desired https://github.com/openzipkin/zipkin-api/blob/master/zipkin2-api.yaml

**Review by codefromthecrypt — APPROVED** [review]
I like the centralization

**Review by zeagord — APPROVED** [review]

File: zipkin-lens/src/prop-types/index.js (hunk 2 above)
Comment (tacigar) [reply zipkin-lens/src/prop-types/index.js:25]: @adriancole 
Sorry, I don't understand what you mean 😭 
What should I do?

Comment (codefromthecrypt) [reply zipkin-lens/src/prop-types/index.js:25]: Add above this line

```
// TODO: Verify which fields we should enforce here, as some are optional per https://github.com/openzipkin/zipkin-api/blob/master/zipkin2-api.yaml
```

Comment (tacigar) [reply zipkin-lens/src/prop-types/index.js:25]: OK
Thank you!

**Comment by tacigar** [comment]
PTAL 🙇

**Review by codefromthecrypt — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
just awaiting travis to finish

**Pull request merged**